pkg/dial/dialer.go

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package dial
+
+import (
+	"context"
+	"net"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+type Resolver interface {
+	// Resolve translates the provided hostname into the corresponding IP address, or
+	// possibly another alias DNS name. An error is returned if no mapping is found.
```

Comment:
Does it make sense to define an explicit "not found" error type, to distinguish between empty lookup and transient failure?

Reply:
Hmm, I'm not actually that sure whether there's a clear distinction between temporary and not found errors, in the sense that even a not found error may be temporary (e.g., the creation event for the given service has not yet been received). Considering that the error possibly returned by a resolver is ignored by the dialer anyway to keep the behavior as transparent as possible (hence just dialing with the original hostname), there's probably little value in distinguishing the type of error.